coderd/httpmw/logger.go

```diff
@@ -35,7 +35,18 @@ func Logger(log slog.Logger) func(next http.Handler) http.Handler {
 				slog.F("start", start),
 			)
 
-			next.ServeHTTP(sw, r)
+			logContext := &RequestLoggerContext{
+				log:     &httplog,
+				written: false,
+				message: r.Method,
+			}
+			defer func() {
+				logContext.WriteLog(r.Context(), sw.Status)
+			}()
+
+			ctx := context.WithValue(r.Context(), logContextKey{}, logContext)
+
+			next.ServeHTTP(sw, r.WithContext(ctx))
```

Comment:
All of the code below should not call `logLevelFn`. I'd replace lines 64-84 with a call to `logContext.WriteLog` instead of using a deferred func perhaps.

Reply:
You'll probably also want to do the `tracing.RunWithoutSpan`